flake.nix

```
@@ -311,10 +311,15 @@
                 paletteGenerator =
                   self.packages.${pkgs.stdenv.hostPlatform.system}.palette-generator;
                 base16 = base16.lib args;
-                homeManagerIntegration.module = self.homeManagerModules.stylix;
+                homeManagerIntegration.module = self.homeModules.stylix;
               };
             }
           ];
         };
-    };
+    }
+    //
+      # Drop this alias after 26.05
+      nixpkgs.lib.optionalAttrs (!nixpkgs.lib.oldestSupportedReleaseIsAtLeast 2511) {
+        homeManagerModules = builtins.warn "stylix: flake output `homeManagerModules` has been renamed to `homeModules`" self.homeModules;
```

Optional, if you plan to backport this change you could do this to prevent warnings on your 24.11 branch:

```suggestion
        homeManagerModules =
          builtins.warnIf
            (nixpkgs.lib.oldestSupportedReleaseIsAtLeast 2411)
            "stylix: flake output `homeManagerModules` has been renamed to `homeModules`"
            self.homeModules;
```

Or `2505` if you don't want the warning to show up until 25.05 is released

(in addition to the `optionalAttrs` you already have)

---

this won't be backported so I don't really feel like it's necessary.